src/core/Akka/Actor/SupervisorStrategy.cs

```diff
@@ -193,10 +190,9 @@ public static SupervisorStrategy DefaultStrategy
     /// </summary>
     public class OneForOneStrategy : SupervisorStrategy
     {
-        private readonly int _maxNumberOfRetries;
-        private readonly int _withinTimeRangeMilliseconds;
-        private readonly Func<Exception, Directive> _decider;
-        private readonly bool _loggingEnabled;
+        public int MaxNumberOfRetries { get; private set; }
+        public int WithinTimeRangeMilliseconds { get; private set; }
+        public IDecider Decider { get; private set; }
 
```

Comment:
this is not a joke, but I just got bitten by this exact issue.

I did add the IEnumerable overload to the DeployableDecider, so there is one for arrays and one for IEnumerable.

```
Result Message:	Newtonsoft.Json.JsonSerializationException : Unable to find a constructor to use for type Akka.Actor.DeployableDecider. A class should either have a default constructor, one constructor with arguments or a constructor marked with the JsonConstructor attribute. Path 'DefaultDirective', line 1, position 75.
Result StackTrace:	
```

So the constructor resolution is apparently unable to decide which ctor is the most specific one to use.